src/solvers/prop/prop.h

```diff
@@ -44,13 +44,20 @@ class propt
 
   virtual void l_set_to(literalt a, bool value)
   {
-    set_equal(a, const_literal(value));
+    if(value)
+      lcnf({a});
+    else
+      lcnf({!a});
   }
 
   void l_set_to_true(literalt a)
-  { l_set_to(a, true); }
+  {
+    lcnf({a});
+  }
   void l_set_to_false(literalt a)
-  { l_set_to(a, false); }
+  {
+    lcnf({!a});
+  }
 
```

Comment:
Careful -- the l_set_to with two arguments is virtual. Someone inheriting `propt` might hope that they can get intercept the calls to `l_set_to_false` and `l_set_to_true` as well.

Unfortunately this is a bad case of a mixture of interface and implementation.

Reply:
+1 what @kroening said.

Reply:
Finally done.